service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EncounterComponentsMapper.java

```diff
@@ -46,24 +48,29 @@
     private static final String NOT_IMPLEMENTED_MAPPER_PLACE_HOLDER = "<!-- %s/%s -->";
     private static final boolean IS_NESTED = false;
 
-    private final Map<ResourceType, Function<Resource, String>> encounterComponents = ImmutableMap.<ResourceType, Function<Resource,
-        String>>builder()
-        .put(ResourceType.List, this::mapListResource)
-        .put(ResourceType.ProcedureRequest, this::mapProcedureRequest)
-        .put(ResourceType.Observation, this::mapObservation)
-        .put(ResourceType.Immunization, this::mapImmunization)
-        .put(ResourceType.Condition, this::mapCondition)
-        .build();
+    private final Map<ResourceType, Function<Resource, String>> encounterComponents = Map.of(
+        ResourceType.AllergyIntolerance, this::mapAllergyIntolerance,
+        ResourceType.Condition, this::mapCondition,
+        ResourceType.DocumentReference, this::mapDocumentReference,
+        ResourceType.Immunization, this::mapImmunization,
+        ResourceType.List, this::mapListResource,
+        ResourceType.MedicationRequest, this::mapMedicationRequest,
+        ResourceType.Observation, this::mapObservation,
+        ResourceType.ProcedureRequest, this::mapProcedureRequest,
+        ResourceType.ReferralRequest, this::mapReferralRequest);
```

Comment:
I think we can begin to phase out our placeholder now

```suggestion
        ResourceType.ReferralRequest, this::mapReferralRequest,
        ResourceType.DiagnosticReport, this:: mapDefaultNotImplemented);
```